code/modules/antagonists/traitor/components/demoraliser.dm

```diff
@@ -0,0 +1,176 @@
+ * If someone wants to make themselves sad through a camera that's their choice I guess.
+ */
+/datum/proximity_monitor/advanced/demoraliser/proc/on_examine(datum/source, mob/examiner)
+	SIGNAL_HANDLER
+
+	if (ishuman(examiner))
+		on_seen(examiner)
+
+/**
+ * Called when someone is looking at a demoralising object.
+ * Applies a mood if they are conscious and don't already have it.
+ * Different moods are applied based on whether they are an antagonist, authority, or 'other' (presumed crew).
+ *
+ * Arguments
+ * * viewer - Whoever is looking at this.
+ */
+/datum/proximity_monitor/advanced/demoraliser/proc/on_seen(mob/living/carbon/human/viewer)
+	if (!viewer.mind)
+		return
+	// If you're not conscious you're too busy or dead to look at propaganda
+	if (viewer.stat != CONSCIOUS)
+		return
+	if (has_poster_mood_already(viewer))
+		return
+
+	if (is_special_character(viewer))
+		to_chat(viewer, span_notice("[moods.antag_notification]"))
+		SEND_SIGNAL(viewer, COMSIG_ADD_MOOD_EVENT, moods.mood_category, moods.antag_mood)
+	else if (viewer.mind.assigned_role.departments_bitflags & (DEPARTMENT_BITFLAG_SECURITY|DEPARTMENT_BITFLAG_COMMAND))
+		to_chat(viewer, span_notice("[moods.authority_notification]"))
+		SEND_SIGNAL(viewer, COMSIG_ADD_MOOD_EVENT, moods.mood_category, moods.authority_mood)
+	else
+		to_chat(viewer, span_notice("[moods.crew_notification]"))
+		SEND_SIGNAL(viewer, COMSIG_ADD_MOOD_EVENT, moods.mood_category, moods.crew_mood)
+
+	// You don't get points for looking at your own stuff.
+	if (viewer != owner)
+		SEND_SIGNAL(host, moods.mood_signal, owner)
+
+/**
+ * Returns true if the viewer already has been given feelings, false if they haven't.
+ *
+ * Arguments
+ * * viewer - Whoever just saw the parent.
+ */
+/datum/proximity_monitor/advanced/demoraliser/proc/has_poster_mood_already(mob/living/carbon/human/viewer)
+	var/datum/component/mood/mood = viewer.GetComponent(/datum/component/mood)
+	for(var/i in mood.mood_events)
+		var/datum/mood_event/moodlet = mood.mood_events[i]
+		if (moodlet.category == moods.mood_category)
+			return TRUE
+
+	return FALSE
+
+/**** Moods ****/
+
+/// Mood application categories for this objective
+/// Used to reduce duplicate code for applying moods to players based on their state
+/datum/demoralise_moods/
+	/// Mood category to apply to moods
+	var/mood_category
+	/// Event signal to send when mood is applied
+	var/mood_signal = COMSIG_DEMORALISING_EVENT
+
+	/// Text to display to an antagonist upon receiving this mood
+	var/antag_notification
+	/// Mood datum to apply to an antagonist
+	var/datum/mood_event/antag_mood
+	/// Text to display to a crew member upon receiving this mood
+	var/crew_notification
+	/// Mood datum to apply to a crew member
+	var/datum/mood_event/crew_mood
+	/// Text to display to a head of staff upon receiving this mood
+	var/authority_notification
+	/// Mood datum to apply to a head of staff or security
+	var/datum/mood_event/authority_mood
```

Comment:
these are typepaths right? not datums?

Reply:
I am not sure how I am supposed to format this if not this way, the idea here is to have one instance of this datum for each objective type, and each of those has different `datum/mood_event` settings for the three kinds of mood it applies based on your crew type.